packages/mobile/src/home/WalletHome.tsx

```diff
@@ -116,6 +128,16 @@ export class WalletHome extends React.Component<Props, State> {
     // Waiting 1/2 sec before triggering to allow
     // rest of feed to load unencumbered
     setTimeout(this.tryImportContacts, 500)
+
+    this.setState({
```

Comment:
I don't think setting state is necessary here. Generally speaking, you only want to set state when some sort of user action will necessitate a re-render of the component.

In this situation, a change in balance is already going to trigger a re-render because `balances` is being mapped to props. Therefore, there's no benefit in defining your condition in state vs just doing it a method (or defining it as a `const` if this were a functional component) and runs the risk of triggering excessive re-renders.

I would just move this display logic into `shouldShowCashInBottomSheet`

Reply:
I wasn't sure whether we need to calculate balance after `  this.props.refreshAllBalances()` ?

Reply:
I'm not sure what you mean by this. If `refreshBalances` results in a change of state for `balances`, then the app will re-render with the new balances. A re-render would trigger `shouldShowCashInBottomSheet` to execute with the new state.

Setting state in this instance yields no benefits as far as I can see